core/initial.py

```diff
@@ -243,31 +243,32 @@ def do_workspaces(self, params):
 	#           MODULES 			//
 	# ////////////////////////////////
 
-	def alert_results(self, output, prefix=''):
+	def alert_results(self, output, prefix='\t',depth:int = 0,color='N'):
+		if output == [] or output == {}:
+			if depth != 0:
+				self.output('Without result')
+		
 		if isinstance(output, dict):
-			for key in output:
-				self.alert(f"{prefix}{key.upper()}")
-				if isinstance(output[key], dict):
-					self.alert_results(output[key], prefix='\t')
-				elif isinstance(output[key], list):
-					self.alert_results(output[key], prefix='\t')
-				else:
-					self.output(f"\t{output[key]}", 'G')
+			for key, value in output.items():
+				if isinstance(value, list) or isinstance(value, dict):
+					self.alert(f"{prefix*depth}{key.upper()}")
+					self.alert_results(value, prefix=prefix, depth=depth+1, color="G")
+				else :
+					value = value.strip().replace('\n', ' ').replace('\\x', ' ') if isinstance(value, str) else value
+					self.output(f"{prefix*depth}{key.upper()} : {value}", color)
+
 		elif isinstance(output, list):
-			if output == []:
-				self.output('Without result')
-			else:
-				for value in output:
-					self.output(f"\t{value}", 'G')
+			for key in output:
+				self.alert_results(key, prefix='\t', depth=depth, color="G")
+		
 		else:
-			self.output(output)
+			output = output.strip().replace('\n', ' ').replace('\\x', ' ')
+			self.output(f"{prefix*depth}{output}", color)
 
-	def thread(self, function, thread_count, engines, q, q_formats, limit, count, sources):
-		threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=thread_count)
-		futures = (threadpool.submit(
-			function, self, name, q, q_formats, limit, count) for name in engines if name in sources)
-		for _ in concurrent.futures.as_completed(futures):
-			pass
+	def thread(self, *args):
```

Comment:
i am not sure if it's safe to use just `*args`. Maybe `**args` could be better

Reply:
using (**args) can cause errors on calling from other pre-written modules 
due to key-value pairs format👍
because most of the modules I checked use *args format